enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -1007,31 +1007,44 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 			return err
 		}
 
+		if op.GetName() == "" {
+			return status.InvalidArgumentError("Operation.name must be non-empty")
+		}
 		mu.Lock()
 		lastOp = op
 		taskID = op.GetName()
 		stage = operation.ExtractStage(op)
-		if taskID != "" {
-			ctx = log.EnrichContext(ctx, log.ExecutionIDKey, taskID)
-		}
+		ctx = log.EnrichContext(ctx, log.ExecutionIDKey, taskID)
 		mu.Unlock()
 
 		log.CtxDebugf(ctx, "PublishOperation: stage: %s", stage)
 
-		var response *repb.ExecuteResponse
+		actionResourceName, err := digest.ParseUploadResourceName(taskID)
+		if err != nil {
+			return status.WrapErrorf(err, "Failed to parse taskID")
+		}
+		actionResourceName.ToProto().CacheType = rspb.CacheType_AC
```

Comment:
why do we want to mutate a field on a proto returned by ToProto()?

actionResourceNameProto := actionResourceName.ToProto()
actionResourceNameProto.CacheType = rspb.CacheType_AC

or if you're trying to mutation actionResourceName (the struct) directly, is there a public field or setter we could use?